front/components/assistant/Sharing.tsx

```diff
@@ -0,0 +1,331 @@
+    confirmText: "Move to Shared",
+    variant: "primary",
+    showUsage: true,
+  };
+
+  let confirmationModalData: ConfirmationModalDataType | null = {
+    title: "",
+    text: "",
+    confirmText: "",
+    variant: "primary",
+  };
+
+  if (requestNewScope) {
+    confirmationModalData =
+      requestNewScope === "published" && initialScope === "workspace"
+        ? companyToSharedModalData
+        : SCOPE_INFO[requestNewScope].confirmationModalData;
+  }
+
+  const allowedToChange =
+    !disabled &&
+    // never change global assistant
+    initialScope !== "global" &&
+    // only builders can change company assistants
+    (isBuilder(owner) || initialScope !== "workspace");
+
+  return (
+    <div>
+      {requestNewScope && confirmationModalData && (
+        <ScopeChangeModal
+          show={requestNewScope !== null}
+          confirmationModalData={confirmationModalData}
+          usageText={confirmationModalData.showUsage ? usageText : undefined}
+          onClose={() => setModalNewScope(null)}
+          setSharingScope={() =>
+            requestNewScope && setNewScope(requestNewScope)
+          }
+        />
+      )}
+      <DropdownMenu>
+        <DropdownMenu.Button disabled={!allowedToChange}>
+          <div className="group flex cursor-pointer items-center gap-2">
+            <SharingChip scope={newScope} />
+            {allowedToChange && (
+              <IconButton
+                icon={ChevronDownIcon}
+                size="xs"
+                variant="secondary"
+                className="group-hover:text-action-400"
+              />
+            )}
+          </div>
+        </DropdownMenu.Button>
+        <DropdownMenu.Items origin="topRight" width={200}>
+          {Object.entries(SCOPE_INFO)
+            .filter(
+              // can't change to those scopes
+              ([entryScope]) =>
+                entryScope !== "global" &&
+                (isBuilder(owner) || entryScope !== "workspace")
+            )
+            .map(([entryScope, entryData]) => (
+              <DropdownMenu.Item
+                key={entryData.label}
+                label={entryData.label}
+                icon={entryData.icon}
+                selected={entryScope === newScope}
+                onClick={() => {
+                  // no need for modal in the following cases
+                  if (
+                    // assistant is being created
+                    !agentConfigurationId ||
+                    // selection unchanged
+                    entryScope === newScope ||
+                    // selection back to initial state
+                    entryScope === initialScope ||
+                    // the only user of the assistant is the user changing the scope
+                    ((entryScope === "private" || entryScope === "company") &&
+                      assistantInMyList &&
+                      (!agentUsage.agentUsage ||
+                        agentUsage.agentUsage.userCount <= 1))
```

Comment:
Are we sure that this one user is the current user?

Reply:
Yes; there's a change to the usercount logic that the user count should be the number of users that have the assistant in their list. And above we check the assistant is in this user's list

But I changed it to === 1 which is clearer

Reply:
(the change is not active yet though but will be soon, cf ahuna cards)